components/expression/expression-renderer.js

```diff
@@ -116,27 +186,36 @@ genotet.ExpressionRenderer.prototype.layout = function() {
   this.getHeatmapLabelSizes_();
   // Compute the shifting sizes.
   this.profileHeight_ = this.data.options.showProfiles ?
-      this.DEFAULT_PROFILE_HEIGHT : 0;
+    this.DEFAULT_PROFILE_HEIGHT : 0;
 
   this.svgHeatmap_
-    .attr('transform', genotet.utils.getTransform([0, this.profileHeight_]));
+    .attr('transform', genotet.utils.getTransform([0, 0]));
 
   this.heatmapWidth_ = this.canvasWidth_ - this.geneLabelWidth_;
+  if (this.data.options.showGeneLabels) {
+    this.heatmapWidth_ -= this.LABEL_MARGIN_;
+  }
   this.heatmapHeight_ = this.canvasHeight_ - this.profileHeight_ -
-      this.conditionLabelHeight_;
+    this.conditionLabelHeight_;
+  if (this.data.options.showConditionLabels) {
+    this.heatmapHeight_ -= this.LABEL_MARGIN_;
+  }
 
   this.svgHeatmapContent_
-    .attr('transform', genotet.utils.getTransform([this.geneLabelWidth_, 0]));
+    .attr('transform', genotet.utils.getTransform([
+      this.geneLabelWidth_,
+      0
+    ]));
   this.svgConditionLabels_
     .attr('transform', genotet.utils.getTransform([
       this.geneLabelWidth_,
-      this.heatmapHeight_
+      0
     ]));
 };
 
 /** @inheritDoc */
 genotet.ExpressionRenderer.prototype.dataLoaded = function() {
-  this.render();
+  this.render(this.data.options.autoScaleGradient);
```

Comment:
render() is a general function. It shouldn't be accepting parameters like autoScaleGradient, which it could get directly from this.data.options.